rewriter/TEnum.cc

```diff
@@ -123,6 +128,18 @@ vector<ast::ExpressionPtr> processStat(core::MutableContext ctx, ast::ClassDef *
     //
     // So we're good to process this thing as a new T::Enum value.
 
+    core::TypePtr serializeType = core::Types::untypedUntracked();
```

Comment:
Maybe we should blame this to a magic symbol?

Reply:
Alternatively, you could use `nullptr` here. I think it's the case that this type should never show up anywhere from this rewriter pass, and one way to make sure of that would be to use `nullptr`, where the code would simply crash if we forgot to account for the `nullptr` case.

Reply:
You're also using both `bottom` and `untyped` as a signaling value. Maybe that's another indication we should just use `nullptr` for both cases.

Reply:
The reason I left it without a blame is because I just don't emit the `serialize` method if the inferred return type is `T.untyped`. As a sidenote, I think even if we added a blame, and used the `T.untyped` with blame to construct the new method, the blame would be lost once we pull the `Symbol` out of the `TypePtr` to generate the AST.

The reason I used `bottom` and `untyped` as signaling values is that they work well with how I'm `any`ing the types together. `any(x, untyped) == untyped`, which works when any of the variants are not literals, and `any(x, bottom) = x`, which works well as a default/ignore value. Using `nullptr` for both cases would make the code a bit more verbose I think, because then I couldn't just pass them to `any` and have it deal with them for me.

I'm happy to change it use `nullptr` if you think it's better to be explicit here, maybe this is a bit too magical.